addons/pos_self_order_razorpay/controllers/orders.py

```diff
@@ -0,0 +1,69 @@
+from odoo import http
+from odoo.http import request
+from odoo.addons.pos_self_order.controllers.orders import PosSelfOrderController
+from werkzeug.exceptions import Unauthorized
+
+class PosSelfOrderControllerRazorpay(PosSelfOrderController):
+    @http.route("/pos-self-order/razorpay-fetch-payment-status/", auth="public", type="json", website=True)
+    def razorpay_payment_status(self, access_token, order_access_token, payment_data, payment_method_id):
+        pos_config, table = self._verify_authorization(access_token, "", False)
```

Comment:
```suggestion
        pos_config = self._verify_pos_config(access_token)
```
We don't have any use for the table. I guess `_verify_pos_config` would work well here